opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultProvisionService.java

```diff
@@ -1016,6 +1016,10 @@ protected OnmsNode doUpdate(final OnmsNode dbNode) {
                 final boolean changed = handleCategoryChanges(dbNode);
 
                 dbNode.mergeNodeAttributes(node, accumulator);
+                node.getAssetRecord().setId(dbNode.getAssetRecord().getId());
```

Comment:
Does this have any impact on the existing behavior?

Reply:
I didn't see any side-effects yet.

Reply:
Please revisit this and ensure that it is actually working.
This will cause issues if:
* if `node.getAssetRecord().getId()` != `dbNode.getAssetRecord().getId()`
We will most likely end up with an orphaned asset record
Also check, that one side can be null while the other side cannot be null. We will end up with an orphaned asset record as well.